### PR TITLE
feat(finance): add nav buttons

### LINKED ISSUE #111: Finance: allow date navigation
This is a
[ ] bug
[x] feature request
[ ] documentation change
[ ] build/deploy enhancement request

Description:
* Allow Prev month, Prev day, Next day, and Next month navigation using buttons at the bottom of the spreadsheet
* Date headers and all actual/planned items in the spreadsheet has to move left or right on click of these buttons

image

### PR DESCRIPTION
**Pull request by priyankajayaswal1**
fix #111

This PR Fixes : #111 
Screenshot: (if applicable):  
image
Description of the change:

### CODE REVIEW

File: lib/date-helper.tsx [hunk 1]
```diff
@@ -21,12 +21,22 @@ export function getFormattedDate(date: string)
...
+}
+
+export function dateAfterDays(date: Moment.Moment, days: number)
+{
+    return  date.add(days, 'days')
+}
+
+export function dateBeforeDaysFromToday(days: number)
 {
     return Moment().subtract(days, 'days')
```
Comment (souvikbasu) [inline lib/date-helper.tsx:36]: reuse dateBeforeDays and send date param as today where today = Moment()

File: lib/date-helper.tsx [hunk 2]
```diff
@@ -21,12 +21,22 @@ export function getFormattedDate(date: string)
...
 }
 
-export function dateAfterDays(days: number)
+export function dateAfterDaysFromToday(days: number)
 {
     return Moment().add(days, 'days')
```
Comment (souvikbasu) [inline lib/date-helper.tsx:41]: Similarly here

File: next.config.js [hunk 3]
```diff
@@ -4,5 +4,6 @@ module.exports = {
     WALLORA_BACKEND_TOTAL_PLANNED_AMOUNT_WORKITEMWISE:
       'https://wallora.com/api/stats/total-planned-amount-workitemwise',
     WALLORA_BACKEND_TOTAL_PLANNED_EXPENSES_MONTHWISE: 'https://wallora.com/api/stats/total-planned-amount-monthwise',
```
Comment (souvikbasu) [inline next.config.js:6]: do not need these endpoints anymore

File: pages/finance.tsx [hunk 4]
```diff
@@ -1,13 +1,25 @@
...
+  button: {
+    marginLeft: '1rem',
+  },
+})
 
 function Finance() {
+  const classes = useStyles()
   const { user, isLoading, error } = useUser()
+  const [values, setValues] = useState({
+    startDate: dateBeforeDaysFromToday(3),
```
Comment (souvikbasu) [inline pages/finance.tsx:20]: do not hardcode 3 days. Find out how many days will fit in the given width

File: pages/finance.tsx [hunk 5]
```diff
@@ -16,11 +28,74 @@ function Finance() {
...
+          <div className="text-center block relative w-full bottom-0 m-4">
+            <Button
+              size="large"
+              type="submit"
+              variant="contained"
+              color="primary"
+              className={classes.button}
+              onClick={() =>
+                setValues({
+                  startDate: dateBeforeDays(values.startDate, 30),
```
Comment (souvikbasu) [inline pages/finance.tsx:45]: handle months with 31 and 28, 29 days as well. IT should always show 1st of the previous month

**Review by souvikbasu — CHANGES_REQUESTED** [review]

File: pages/finance.tsx (hunk 4 above)
Comment (priyankajayaswal1) [reply pages/finance.tsx]: planning to do as seperate issue

Comment (priyankajayaswal1) [reply pages/finance.tsx]: abstracted it out as dateHeaderCount though.

**Review by souvikbasu — APPROVED** [review]

**Pull request merged**